test/library/pod_same_revision.go

```diff
@@ -12,6 +13,24 @@ import (
 	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
+const revisionLabel = "revision"
+
+func WaitForPodsToStabilizeOnTheSameNewRevision(t LoggingT, podClient corev1client.PodInterface, podLabelSelector string, waitForRevisionSuccessThreshold int, waitForRevisionSuccessInterval, waitForRevisionPollInterval, waitForRevisionTimeout time.Duration) error {
+	pods, err := podClient.List(context.TODO(), metav1.ListOptions{LabelSelector: podLabelSelector})
+	if err != nil {
+		return fmt.Errorf("failed to list pods with selector '%v': %w", podLabelSelector, err)
+	}
+
+	currentRevision, err := getCurrenRevision(revisionLabel, pods.Items)
+	if err != nil {
+		return fmt.Errorf("failed to get current revision for pods with selector '%v': %w", podLabelSelector, err)
+	}
+
+	return wait.Poll(waitForRevisionPollInterval, waitForRevisionTimeout, mustSucceedMultipleTimes(waitForRevisionSuccessThreshold, waitForRevisionSuccessInterval, func() (bool, error) {
```

Comment:
how should we set the suitable waitForRevisionSuccessThreshold and  waitForRevisionSuccessInterval , it is a mysterious number